metapool/metapool.py

```diff
@@ -66,17 +69,28 @@ def read_pico_csv(f, sep='\t', conc_col_name='Sample DNA Concentration'):
     pico_df: pandas DataFrame object
         DataFrame relating well location and DNA concentration
     """
-
-    raw_df = pd.read_csv(f, sep=sep, skiprows=2,
-                         skipfooter=5, engine='python')
-
-    pico_df = raw_df[['Well', '[Concentration]']]
-    pico_df = pico_df.rename(columns={'[Concentration]': conc_col_name})
-
-    # coerce oddball concentrations to np.nan
-    pico_df[conc_col_name] = \
-        pd.to_numeric(pico_df[conc_col_name], errors='coerce')
-
+    if plate_reader == 'Synergy_HT':
+        raw_df = pd.read_csv(f, sep = sep, skiprows=2,
+                             skipfooter=5, engine='python')
+
+        pico_df = raw_df[['Well','[Concentration]']]
+        pico_df = pico_df.rename(columns={'[Concentration]':conc_col_name})
+
+        # coerce oddball concentrations to np.nan
+        pico_df[conc_col_name] = \
+            pd.to_numeric(pico_df[conc_col_name], errors = 'coerce')
+    elif plate_reader == 'SpectraMax_i3x':   
+        raw_df = pd.read_csv(open(f,encoding='utf-16'), sep = sep, skiprows=2,
+                             skipfooter=15, engine='python')
+
+        pico_df = raw_df[['Wells','Concentration']]
+        pico_df = pico_df.rename(columns={'Concentration':conc_col_name,
+                                         'Wells':'Well'})
+
+        # coerce oddball concentrations to np.nan
+        pico_df[conc_col_name] = \
+        pd.to_numeric(pico_df[conc_col_name], errors = 'coerce')
+    
```

Comment:
How does this look to you @RodolfoSalido. Looking at the code I think we can take advantage of Python's default value for [`encoding`](https://docs.python.org/3/library/functions.html#open) in the `open` function. The other key thing is that `rename` will [ignore keys](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rename.html) (see the errors argument) not in the table so we can enter all column names.

I also added a check to raise an exception if the plate reader is not a known value. Can you add a test to check that the exception is raised? For some examples search for `assertRaises` in the test_metapool.py file.

```suggestion
    if plate_reader == 'Synergy_HT':
        encoding, skipfooter = None, 5
    elif plate_reader == 'SpectraMax_i3x':
        encoding, skipfooter = 'utf-16', 15
    else:
        raise ValueError("Invalid plate reader %s" % plate_reader)

    pico_df = pd.read_csv(open(f, encoding=encoding), sep=sep, skiprows=2,
                          skipfooter=skipfooter, engine='python')

    # synergy's concentration column is "Concentration", spectramax's is
    # [Concentration]. Rename will ignore any labels not in the dataframe so
    # only one of the two label updates should happen
    pico_df.rename({'Concentration': conc_col_name,
                    '[Concentration]': conc_col_name,
                    'Wells': 'Well'}, inplace=True)

    pico_df = pico_df[['Well', conc_col_name]].copy()

    # coerce oddball concentrations to np.nan
    pico_df[conc_col_name] = \
        pd.to_numeric(pico_df[conc_col_name], errors='coerce')

    return pico_df
```

Reply:
looks good to me.